Rtl support

GianBe opened the pull request:
Possible placeholder solution untill facebook releases their unicode detection tool

GianBe committed:
Test rtl languages support (not formatted)

GianBe committed:
Moved bidi detection to key-utils.js

GianBe committed:
Updated RTL_RANGES to the unicode 8.0.0 standard

GianBe committed:
Corrected rtl ranges and formatted the code

GianBe committed:
Changed how the text direction change and added a dir prop

GianBe commented on src/FacetedTokenInput.js:
```diff
@@ -233,15 +258,30 @@ export default class FacetedTokenInput extends Component {
+        });
+      }
+    }
+
```
In this `if()` the direction of the text changes when it's not specified in the props (in this case the initial direction is set by your default language) and dirChange is set to false so the direction doesn't always try to change

GianBe commented on src/FacetedTokenInput.js:
```diff
@@ -214,6 +231,14 @@ export default class FacetedTokenInput extends Component {
+        this.setState({
+          dirChange: true
+        });
+      }
```
If the input is totally empty (no tokens and no text) the direction can change again. 

This isn't the best place to check if the input is empty since this function is on a loop (even if it's on a loop setState is never called more than 1 time) but it prevents at least one small error.

If the detection is placed inside the onChange or the onBackspace function:
- `onChange`: if the last token in the input is deleted the function doen't get called even if the input is empty
- `onBackspace`: if the last token is deleted the tokens.lenght doesn't get updated immediately leaving the direction locked even if the input is empty.

yuchi commented on src/FacetedTokenInput.js:
```diff
@@ -233,15 +258,30 @@ export default class FacetedTokenInput extends Component {
       showDropDown: true,
       selectedSectionIndex: -1,
-      selectedIndex: -1
+      selectedIndex: -1,
```
trailing comma here :)

GianBe committed:
Fixed linting errors (npm run lint) on both files

Fixed trailing commas and unused variables also changed:
dir ? dir : textDirection
to:
dir || textDirection
since it was an unnecessary ternary

GianBe committed:
Updated usage with new 'dir' prop

yuchi committed:
Fire forgotten onChange cases

yuchi committed:
Simplified rules for rtl/ltr checking

yuchi committed:
Smaller optimizations for rtl/ltr checks
